PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -747,16 +747,21 @@ AuthenticateCapsule (
   }
 
   if (Header->ImageOffset >= FwSize || Header->ImageOffset + Header->ImageSize >= FwSize) {
-    DEBUG ((DEBUG_ERROR, "Invalid capsule: ImageOffset=0x%x, ImageOffset=0x%x\n", Header->ImageOffset, Header->ImageOffset));
+    DEBUG ((DEBUG_ERROR, "Invalid capsule: ImageOffset=0x%x, ImageSize=0x%x\n", Header->ImageOffset, Header->ImageSize));
     return EFI_INVALID_PARAMETER;
   }
 
-  PubKeyHdr       = (PUB_KEY_HDR *) (FwImage + Header->PubKeyOffset);
-  SignatureHdr    = (SIGNATURE_HDR *) (FwImage + Header->SignatureOffset);
-  Status = DoRsaVerify (FwImage, Header->SignatureOffset, HASH_USAGE_PUBKEY_FWU, SignatureHdr, PubKeyHdr, PcdGet8(PcdCompSignHashAlg), NULL, NULL);
-  if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, "Image verification failed, %r!\n", Status));
-    return EFI_SECURITY_VIOLATION;
+  if (!FeaturePcdGet (PcdVerifiedBootEnabled)) {
```

Comment:
When verified boot is turned off in SBL, Fwupdate doesn't need verification. 
AuthenticateCapsule should be kept under feature flag PcdVerifiedBootEnabled.

Reply:
Yes, I was thinking about this, since the AuthenticateCapsule() includes some file size checks. So added the evaluation on RsaVerify only. But it's good to keep the evaluation on AuthenticateCapsule(), I will move the code over.